src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -788,6 +791,12 @@ public int RunTest(string executable, string outputFile, string errorFile, strin
                                 executable, timeout, (environmentVar != null) ? " from variable " : "", (environmentVar != null) ? TIMEOUT_ENVIRONMENT_VAR : "",
                                 startTime.ToString(), endTime.ToString());
 
+                        Console.WriteLine("Timed Out with Active Processes:");
```

Comment:
The log above said we timed out so I don't think we need to repeat that again. Perhaps output like this:
```
cmdLine: blablabla Timed Out (timeout in milliseconds: XYZ, start: A, end: B)
Collecting diagnostic information...
Snapshot of processes currently running:
   ID    ParentId  CmdLine
   145   9001      Foo.exe arg1 arg2
   1458  145       Blah.exe something something
   10632 2056      corerun.exe SomeTest.dll --whatever arg more_args
   ...

Reply:
Do we already have a straightforward way to grab the ParentIds and the CommandLine args of the process?

Reply:
I was hoping that either the existing enumeration code that handles GetChildren() would have it or you could get it from Process.GetProcesses(). If neither of them provide an easy option feel free to simplify it as much as needed. Maybe it would just be ID and ProcessName as a simpler alternative.

Reply:
The platform specific implementation for `GetChildren` are also in this file https://github.com/dotnet/runtime/blob/eed57c3ede7832d80b34bdf70ac2b66b44e9e437/src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs#L112-L228

I think just validating whether or not the subprocess timing out is a part of that list would be a good initial check, and if we do need to extend/add `GetParent`, that can be added in a separate PR.